### ISSUE TITLE
Navbar w/o navbar-brand breaks with navbar-toggler-right(-left)

### ISSUE THREAD

**Issue by iatek** [twbs/bootstrap#21846]
When you create a navbar with only a `navbar-nav` component (no `navbar-brand`), and then use `navbar-toggler-right` or `navbar-toggler-left`, the entire navbar height collapses when the toggler is shown. 

http://www.codeply.com/go/smB7eC4qYc

A workaround I've found is to use an `navbar-text` placeholder...

```
<nav class="navbar navbar-toggleable-md navbar-light bg-faded">
  <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#navbarToggler1" aria-controls="navbarToggler1" aria-expanded="false" aria-label="Toggle navigation">
    <span class="navbar-toggler-icon">&nbsp;</span>
  </button>
  <span class="navbar-text">&nbsp;</span>
  <div class="collapse navbar-collapse" id="navbarToggler1">
    <ul class="navbar-nav mr-auto mt-2 mt-md-0">
      <li class="nav-item active">
        <a class="nav-link" href="#">Home <span class="sr-only">(current)</span></a>
      </li>
      <li class="nav-item">
        <a class="nav-link" href="#">Link</a>
      </li>
      <li class="nav-item">
        <a class="nav-link disabled" href="#">Disabled</a>
      </li>
    </ul>
    <form class="form-inline my-2 my-lg-0">
      <input class="form-control mr-sm-2" type="text" placeholder="Search">
      <button class="btn btn-outline-success my-2 my-sm-0" type="submit">Search</button>
    </form>
  </div>
</nav>
```

**Comment by RusinovAnton** [comment]
Why don't we just use `align-self` instead of `position absolute;`?

**Comment by AndrewMorris-scsu** [comment]
Until this gets resolved, I found a workaround. 

Within the toggled div, I.E. "navbar-collapse". Add a col class to the div, see line 6 of the example below. 

Here is the codeply showing the fix: [http://www.codeply.com/go/knGruhdOp1](url)

Here is the code: 
```html
<nav class="navbar navbar-toggleable-md navbar-light bg-faded">
  <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#navbarToggler1" aria-controls="navbarToggler1" aria-expanded="false" aria-label="Toggle navigation">
    <span class="navbar-toggler-icon">&nbsp;</span>
  </button>
  <span class="navbar-text">&nbsp;</span>
  <div class="collapse navbar-collapse col-12" id="navbarToggler1">
    <ul class="navbar-nav mr-auto mt-2 mt-md-0">
      <li class="nav-item active">
        <a class="nav-link" href="#">Home <span class="sr-only">(current)</span></a>
      </li>
      <li class="nav-item">
        <a class="nav-link" href="#">Link</a>
      </li>
      <li class="nav-item">
        <a class="nav-link disabled" href="#">Disabled</a>
      </li>
    </ul>
    <form class="form-inline my-2 my-lg-0">
      <input class="form-control mr-sm-2" type="text" placeholder="Search">
      <button class="btn btn-outline-success my-2 my-sm-0" type="submit">Search</button>
    </form>
  </div>
</nav>
```

Hope this helps!

**Comment by iatek** [comment]
@AndrewMorris-scsu - the `col-12` isn't a workaround. The codeply demo only works because it includes the original workaround I mentioned above..

`<span class="navbar-text">&nbsp;</span>`

**Comment by AndrewMorris-scsu** [comment]
Ah, Sorry about that!
I'm on cell now, and I'll update the post later. I am working on another workaround that uses a transparent brand icon. 
At least that way you are not relying on nbsp.

**Comment by flakolefluk** [comment]
my temp solution (assuming default font sizes):

custom.css
`.no-brand{
margin-right: auto;
min-height:2.6rem;
}
`

then on the navbar-brand use the no-brand class
`<a class="navbar-brand no-brand" href="#"></a>`

**Comment by Willmforeman** [comment]
Adding the property ```align-self: flex-end;``` to the ```.navbar-toggler``` class is the simplest workaround I have found thus far.

**Comment by Wfore0990** [comment]
Changing the ```align-self``` property for the ```.navbar-toggler``` class is the simplest workaround I have found so far. 

Use ```align-self: flex-start;``` for left-side alignment or ```align-self: flex-end;``` for right-side alignment.

CSS example (right-side alignment):
```
.navbar-toggler {
     align-self: flex-end; 
}
```

**Comment by mdo** [comment]
Fix coming in #22230.
